### PR TITLE
Changelog entry for v0.7.0

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds changelog entry for v0.7.0.

**Preview:**

Previous release at https://github.com/GenericMappingTools/pygmt/pull/1874.

Changelog made by following the instructions at [https://github.com/GenericMappingTools/pygmt/blob/main/doc/maintenance.md#updating-the-changelog](https://github.com/GenericMappingTools/pygmt/blob/main/doc/maintenance.md?rgh-link-date=2022-04-09T01%3A26%3A15Z#updating-the-changelog)

TODO:

- [x] Copy the draft release note from https://github.com/GenericMappingTools/pygmt/releases to `doc/changes.md`
- [x] Update the DOI badge
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt and `CITATION.cff`
- [x] Add the version to the compatibility table https://github.com/GenericMappingTools/pygmt#compatibility-with-gmtpythonnumpy-versions
- [x] Add highlights
- [x] Update with any remaining PRs merged before release
- [x] Edit the change list to remove any trivial changes (updates to the README, typo fixes, CI configuration, etc)
- [x] Edit the list of people who contributed to the release, linking to their GitHub account. Sort their names by the number of commits made since the last release (e.g., use `git shortlog HEAD...v0.1.2 -sne`).

Relates to #1966

### CODE REVIEW

File: doc/changes.md [hunk 1]
```diff
@@ -1,5 +1,76 @@
...
+* Wrap binstats ([#1652](https://github.com/GenericMappingTools/pygmt/pull/1652))
+* Wrap filter1d ([#1512](https://github.com/GenericMappingTools/pygmt/pull/1512))
+* Wrap dimfilter ([#1492](https://github.com/GenericMappingTools/pygmt/pull/1492))
+
+### Enhancements
+
+* Support passing data in numpy int8, int16, uint8 and uint16 dtypes to GMT ([#1963](https://github.com/GenericMappingTools/pygmt/pull/1963))
+* inset: Add aliases for R and J and fix two examples ([#1931](https://github.com/GenericMappingTools/pygmt/pull/1931))
+* basemap: Plotting frames if required parameters are not given ([#1909](https://github.com/GenericMappingTools/pygmt/pull/1909))
+* Added alias for F in basemap ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
```
Comment (seisman) [inline doc/changes.md:22]:
```suggestion
* basemap: Added alias for F ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
```

File: doc/changes.md [hunk 2]
```diff
@@ -1,5 +1,76 @@
...
+* Add a sample dataset MaunaLoa_CO2 ([#1961](https://github.com/GenericMappingTools/pygmt/pull/1961))
```
Comment (seisman) [inline doc/changes.md:23]:
```suggestion
* Add a sample dataset maunaLoa_co2 ([#1961](https://github.com/GenericMappingTools/pygmt/pull/1961))
```

File: doc/changes.md [hunk 3]
```diff
@@ -1,5 +1,76 @@
...
+* Add a sample dataset notre_dame_topography ([#1920](https://github.com/GenericMappingTools/pygmt/pull/1920))
+* Add earth_relief_holes to load_sample_data() ([#1921](https://github.com/GenericMappingTools/pygmt/pull/1921))
```
Comment (seisman) [inline doc/changes.md:25]:
```suggestion
* Add a sample dataset earth_relief_holes ([#1921](https://github.com/GenericMappingTools/pygmt/pull/1921))
```

File: doc/changes.md [hunk 4]
```diff
@@ -1,5 +1,76 @@
...
+
+### Bug Fixes
+
+* grdtrack: Fix the bug when profile is given ([#1867](https://github.com/GenericMappingTools/pygmt/pull/1867))
+* Fix the grid accessor (grid registration and type) for 3D grids ([#1913](https://github.com/GenericMappingTools/pygmt/pull/1913))
+
+### Documentation
+
+* Add instructions to install PyGMT using mamba ([#1967](https://github.com/GenericMappingTools/pygmt/pull/1967))
+* Improvement of two gallery examples regarding categorical colormaps ([#1934](https://github.com/GenericMappingTools/pygmt/pull/1934))
```
Comment (seisman) [inline doc/changes.md:41]:
```suggestion
* Improve two gallery examples regarding categorical colormaps ([#1934](https://github.com/GenericMappingTools/pygmt/pull/1934))
```

File: doc/changes.md [hunk 5]
```diff
@@ -1,5 +1,76 @@
...
+* Add inline example to dimfilter ([#1956](https://github.com/GenericMappingTools/pygmt/pull/1956))
+* Add inline example to surface ([#1953](https://github.com/GenericMappingTools/pygmt/pull/1953))
+* Add inline example for grdfill ([#1954](https://github.com/GenericMappingTools/pygmt/pull/1954))
```
Comment (seisman) [inline doc/changes.md:44]:
```suggestion
* Add inline example to grdfill ([#1954](https://github.com/GenericMappingTools/pygmt/pull/1954))
```

File: doc/changes.md [hunk 6]
```diff
@@ -1,5 +1,76 @@
...
+* Add inline code examples to contributing guidelines ([#1924](https://github.com/GenericMappingTools/pygmt/pull/1924))
+* Add two new PyGMT tutorials to external resources ([#1971](https://github.com/GenericMappingTools/pygmt/pull/1971))
+* Add thumbnail images to the external resources page ([#1941](https://github.com/GenericMappingTools/pygmt/pull/1941))
+* Add EGU 2022 short course to external resources ([#1935](https://github.com/GenericMappingTools/pygmt/pull/1935))
```
Comment (seisman) [inline doc/changes.md:48]:
```suggestion
* Add two new PyGMT tutorials to external resources page ([#1971](https://github.com/GenericMappingTools/pygmt/pull/1971))
* Add thumbnail images to the external resources page ([#1941](https://github.com/GenericMappingTools/pygmt/pull/1941))
* Add EGU 2022 short course to external resources page ([#1935](https://github.com/GenericMappingTools/pygmt/pull/1935))
```

Comment (weiji14) [reply doc/changes.md:48]: We could probably turn these 'external resources' page bullet points into a highlight for v0.7.0.

**Review by weiji14 — COMMENTED** [review]
Need to add @yvonnefroehlich and @andrebelem's names to CITATION.cff too

File: doc/changes.md [hunk 7]
```diff
@@ -1,5 +1,76 @@
...
+
+* Add instructions to install PyGMT using mamba ([#1967](https://github.com/GenericMappingTools/pygmt/pull/1967))
+* Improve two gallery examples regarding categorical colormaps ([#1934](https://github.com/GenericMappingTools/pygmt/pull/1934))
+* Add inline example to dimfilter ([#1956](https://github.com/GenericMappingTools/pygmt/pull/1956))
+* Add inline example to surface ([#1953](https://github.com/GenericMappingTools/pygmt/pull/1953))
+* Add inline example to grdfill ([#1954](https://github.com/GenericMappingTools/pygmt/pull/1954))
+* Add inline code examples to contributing guidelines ([#1924](https://github.com/GenericMappingTools/pygmt/pull/1924))
+* Add two new PyGMT tutorials to external resources page ([#1971](https://github.com/GenericMappingTools/pygmt/pull/1971))
+* Add thumbnail images to the external resources page ([#1941](https://github.com/GenericMappingTools/pygmt/pull/1941))
+* Add EGU 2022 short course to external resources page ([#1935](https://github.com/GenericMappingTools/pygmt/pull/1935))
```
Comment (seisman) [inline doc/changes.md:48]:
```suggestion
* Add thumbnail images to the external resources page ([#1941](https://github.com/GenericMappingTools/pygmt/pull/1941))
```

**Comment by seisman** [comment]
> Need to add @yvonnefroehlich and @andrebelem's names to CITATION.cff too

I've added @yvonnefroehlich and @andrebelem to `CITATION.cff` (14e4d90f1004781c3c414e9506eba8c825f1ed54). The order are based on the number of commits listed in this page (https://github.com/GenericMappingTools/pygmt/graphs/contributors).

File: CITATION.cff [hunk 8]
```diff
@@ -44,6 +44,14 @@ authors:
   family-names: Materna
   affiliation: US Geological Survey, USA
   orcid: https://orcid.org/0000-0002-6687-980X
+- given-names: Yvonne
+  family-names: Fröhlich
+  affiliation: Karlsruhe Institute of Technology, Germany
+  orcid: https://orcid.org/0000-0002-8566-0619
+- given-names: Andre L.
```
Comment (seisman) [inline CITATION.cff:51]: Please double check that the given name is `Andre L.` or `Andre`.

File: doc/changes.md [hunk 9]
```diff
@@ -1,5 +1,76 @@
...
+* basemap: Added alias for F ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
+* Add a sample dataset maunaLoa_co2 ([#1961](https://github.com/GenericMappingTools/pygmt/pull/1961))
+* Add a sample dataset notre_dame_topography ([#1920](https://github.com/GenericMappingTools/pygmt/pull/1920))
+* Add a sample dataset earth_relief_holes ([#1921](https://github.com/GenericMappingTools/pygmt/pull/1921))
+
+### Deprecations
+
+* Figure.wiggle: Remove parameter 'columns', use 'incols' instead. ([#1977](https://github.com/GenericMappingTools/pygmt/pull/1977))
+* Figure.histogram and pygmt.info: Remove parameter 'table', use 'data' instead ([#1975](https://github.com/GenericMappingTools/pygmt/pull/1975))
+* pygmt.surface: Remove parameter 'outfile', use 'outgrid' instead ([#1976](https://github.com/GenericMappingTools/pygmt/pull/1976))
```
Comment (seisman) [inline doc/changes.md:31]:
```suggestion
* pygmt.surface: Remove parameter 'outfile', use 'outgrid' instead ([#1976](https://github.com/GenericMappingTools/pygmt/pull/1976))
* blockm/contour/plot/plot3d/rose/surface/wiggle: Change the parameter order of data array and input arrays
```

**Review by michaelgrund — APPROVED** [review]

File: CITATION.cff (hunk 8 above)
Comment (maxrjones) [inline CITATION.cff:51]:
```suggestion
- given-names: Andre
```
Based on https://github.com/GenericMappingTools/egu22pygmt/pull/16

File: doc/changes.md [hunk 10]
```diff
@@ -1,5 +1,77 @@
...
+* Update GMT Dev Tests workflow to test on macOS-12 and ubuntu-22.04 ([#1918](https://github.com/GenericMappingTools/pygmt/pull/1918))
+
+### Contributors
+
+* [Dongdong Tian](https://github.com/seisman)
+* [Will Schlitzer](https://github.com/willschlitzer)
+* [Wei Ji Leong](https://github.com/weiji14)
+* [Andre L. Belem](https://github.com/andrebelem)
+* [Yvonne Fröhlich](https://github.com/yvonnefroehlich)
+* [Max Jones](https://github.com/meghanrjones)
```
Comment (maxrjones) [inline doc/changes.md:69]:
```suggestion
* [Max Jones](https://github.com/maxrjones)
```

File: doc/changes.md [hunk 11]
```diff
@@ -1,5 +1,77 @@
...
+
+### New Features
+
+* Wrap binstats ([#1652](https://github.com/GenericMappingTools/pygmt/pull/1652))
+* Wrap filter1d ([#1512](https://github.com/GenericMappingTools/pygmt/pull/1512))
+* Wrap dimfilter ([#1492](https://github.com/GenericMappingTools/pygmt/pull/1492))
+
+### Enhancements
+
+* Support passing data in numpy int8, int16, uint8 and uint16 dtypes to GMT ([#1963](https://github.com/GenericMappingTools/pygmt/pull/1963))
```
Comment (maxrjones) [inline doc/changes.md:21]:
```suggestion
* Support passing data in NumPy int8, int16, uint8 and uint16 dtypes to GMT ([#1963](https://github.com/GenericMappingTools/pygmt/pull/1963))
```

File: doc/changes.md [hunk 12]
```diff
@@ -1,5 +1,77 @@
...
+* inset: Add aliases for R and J and fix two examples ([#1931](https://github.com/GenericMappingTools/pygmt/pull/1931))
```
Comment (maxrjones) [inline doc/changes.md:22]:
```suggestion
* inset: Add region and projection aliases and fix two examples ([#1931](https://github.com/GenericMappingTools/pygmt/pull/1931))
```

File: doc/changes.md [hunk 13]
```diff
@@ -1,5 +1,77 @@
...
+* basemap: Plotting frames if required parameters are not given ([#1909](https://github.com/GenericMappingTools/pygmt/pull/1909))
+* basemap: Added alias for F ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
```
Comment (maxrjones) [inline doc/changes.md:24]:
```suggestion
* basemap: Added box alias for F ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
```

**Review by maxrjones — APPROVED** [review]

File: doc/changes.md [hunk 14]
```diff
@@ -1,5 +1,77 @@
...
+* Add a sample dataset maunaLoa_co2 ([#1961](https://github.com/GenericMappingTools/pygmt/pull/1961))
+* Add a sample dataset notre_dame_topography ([#1920](https://github.com/GenericMappingTools/pygmt/pull/1920))
+* Add a sample dataset earth_relief_holes ([#1921](https://github.com/GenericMappingTools/pygmt/pull/1921))
+
+### Deprecations
+
+* Figure.wiggle: Remove parameter 'columns', use 'incols' instead. ([#1977](https://github.com/GenericMappingTools/pygmt/pull/1977))
+* Figure.histogram and pygmt.info: Remove parameter 'table', use 'data' instead ([#1975](https://github.com/GenericMappingTools/pygmt/pull/1975))
+* pygmt.surface: Remove parameter 'outfile', use 'outgrid' instead ([#1976](https://github.com/GenericMappingTools/pygmt/pull/1976))
+* blockm/contour/plot/plot3d/rose/surface/wiggle: Change the parameter order of data array and input arrays
```
Comment (maxrjones) [inline doc/changes.md:34]:
```suggestion
* blockm/contour/plot/plot3d/rose/surface/wiggle: Change the parameter order of data array and input arrays ([#1978](https://github.com/GenericMappingTools/pygmt/pull/1978))
```

**Review by weiji14 — APPROVED** [review]
Other than Max's suggested changes, looks good to me! Thanks again @seisman for handling the changelog and the release!

File: README.rst [hunk 15]
```diff
@@ -236,7 +236,12 @@ Compatibility with GMT/Python/NumPy versions
       - >=6.3.0
       - >=3.8
       - >=1.19
-    * - `v0.6.1 <https://github.com/GenericMappingTools/pygmt/releases/tag/v0.6.1>`_ (latest release)
+    * - `v0.7.0 <https://github.com/GenericMappingTools/pygmt/releases/tag/v0.7.0>`_ (latest release)
+      - `v0.7.0 Documentation <https://www.pygmt.org/v0.7.0>`_
+      - >=6.3.0
+      - >=3.8
+      - >=1.19
```
Comment (maxrjones) [inline README.rst:243]:
```suggestion
      - >=1.20
```

Comment (weiji14) [reply README.rst:243]: Maybe put this change in #1985, following what was done in previous NEP29 PRs.

Comment (maxrjones) [reply README.rst:243]: Done, but we should leave this suggestion open because that is for the `dev` row and this is for the `v0.7.0` row.

Comment (weiji14) [reply README.rst:243]: Ah yes, let's keep this here then.

File: doc/changes.md [hunk 16]
```diff
@@ -1,5 +1,77 @@
...
+* inset: Add aliases for R and J and fix two examples ([#1931](https://github.com/GenericMappingTools/pygmt/pull/1931))
+* basemap: Plotting frames if required parameters are not given ([#1909](https://github.com/GenericMappingTools/pygmt/pull/1909))
+* basemap: Added alias for F ([#1894](https://github.com/GenericMappingTools/pygmt/pull/1894))
+* Add a sample dataset maunaLoa_co2 ([#1961](https://github.com/GenericMappingTools/pygmt/pull/1961))
+* Add a sample dataset notre_dame_topography ([#1920](https://github.com/GenericMappingTools/pygmt/pull/1920))
+* Add a sample dataset earth_relief_holes ([#1921](https://github.com/GenericMappingTools/pygmt/pull/1921))
+
+### Deprecations
+
+* Figure.wiggle: Remove parameter 'columns', use 'incols' instead. ([#1977](https://github.com/GenericMappingTools/pygmt/pull/1977))
```
Comment (maxrjones) [inline doc/changes.md:31]:
```suggestion
* NEP29: Set minimum required version to NumPy 1.20+ ([#1985](https://github.com/GenericMappingTools/pygmt/pull/1985))
* Figure.wiggle: Remove parameter 'columns', use 'incols' instead. ([#1977](https://github.com/GenericMappingTools/pygmt/pull/1977))
```

File: README.rst [hunk 17]
```diff
@@ -173,12 +173,12 @@ Developers". Feel free to cite our work in your research using the following Bib
                       Quinn, Jamie and
```
Comment (weiji14) [inline README.rst:173]: Forgot to add Andre and Yvonne's names to the BibTeX :sweat_smile:

**Pull request merged**